xdsl/irdl/declarative_assembly_format.py

```diff
@@ -723,6 +727,85 @@ def print(self, printer: Printer, state: PrintingState, op: IRDLOperation) -> No
             state.should_emit_space = True
 
 
+@dataclass(frozen=True)
+class RegionVariable(VariableDirective, OptionallyParsableDirective):
+    """
+    A region variable, with the following format:
+      region-directive ::= dollar-ident
+    The directive will request a space to be printed after.
+    """
+
+    def parse(self, parser: Parser, state: ParsingState) -> None:
```

Comment:
Oh wait: Only the OptionalRegionVariabl needs to implement it, not this one, indeed :+1: Good catch!

Reply:
Sorry I wasn't clear, I meant that I think `parse` gets a default implementation from the definition of `parse_optional`. I think you could concoct a weird example where you could still use a region in an optional group even if it's non optional

Reply:
You're right, it does get a default implementation

Reply:
> Sorry I wasn't clear, I meant that I think `parse` gets a default implementation from the definition of `parse_optional`. I think you could concoct a weird example where you could still use a region in an optional group even if it's non optional

You're right; just for full context: xDSL doesn't implement that case yet (optional group's else-elements) so a lot of other similar declarative variables do not implement it either.

So for the sake of this PR, both removing it or keeping it would be fine :slightly_smiling_face:

If keeping the optional, using the default `parse` implementation if it is working would be nice indeed!

Reply:
Keeping just parse_optional and allowing the default parse implementaion to be used